internal/logger/target/http/http.go

```diff
@@ -361,32 +361,29 @@ func (h *Target) SendFromStore(key string) (err error) {
 	return h.store.Del(key)
 }
 
-// Send log message 'e' to http target.
-// If servers are offline messages are queued until queue is full.
+// Send the log message 'entry' to the http target.
+// Messages are queued in the disk if the store is enabled
 // If Cancel has been called the message is ignored.
 func (h *Target) Send(ctx context.Context, entry interface{}) error {
+	if atomic.LoadInt32(&h.status) == statusClosed {
+		return nil
+	}
```

Comment:
moving this here means that we won't be able to store logs if the remote status is offline or closed.

Reply:
> moving this here means that we won't be able to store logs if the remote status is offline or closed.

well, if it is closed, then we don't have to store anything (closed means we explicity asked for the logger to stop working)

Reply:
this is also preserved in case of situations where queue_dir is set and target is closed, we can re-enable it and it would requeue all of the past events.

Reply:
> this is also preserved in case of situations where queue_dir is set and target is closed, we can re-enable it and it would requeue all of the past events.

What do you mean by target closed here ? in the close the state closed is only when you call target.Cancel() (there is another state which is called offline which has another meaning)

Reply:
correct target can be renabled later after a Cancel() @vadmeste the code is present to ensure we always preserve events.

Reply:
It can argued but this is breaking something that someone might be already relying on.